### PR TITLE
Backport udp connect support from master

### PR DESCRIPTION
**Pull request by santigimeno**
Refs: https://github.com/libuv/libuv/pull/1872#issuecomment-464080195

I don't know the policy wrt the metadata in these cases. For the moment I have kept the metadata from the original commits.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1259/

### CODE REVIEW

**Comment by santigimeno** [comment]
If this is merged, we'll need to modify [this nodejs test](https://github.com/nodejs/node/blob/master/test/parallel/test-dgram-address.js#L74_L81) when upgrading libuv there.

File: docs/src/udp.rst [hunk 1]
```diff
@@ -150,6 +150,44 @@ API
...
+
+    :param handle: UDP handle. Should have been initialized with
+        :c:func:`uv_udp_init`.
+
+    :param addr: `struct sockaddr_in` or `struct sockaddr_in6`
+        with the address and port to associate to.
+
+    :returns: 0 on success, or an error code < 0 on failure.
+
+    .. versionadded:: 2.0.0
```
Comment (saghul) [inline docs/src/udp.rst]: 1.27 I think?

**Comment by santigimeno** [comment]
Updated with documentation fix.

**Comment by cjihrig** [comment]
Landed in af31d014...90415a33. Thanks!

**Comment by 1st1** [comment]
Amazing! Thank you!

**Pull request closed without merging**